src/context/example.rs

```diff
@@ -0,0 +1,53 @@
+use report::example::ExampleReport;
+
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum ExampleLabel {
+    It,
+    Example,
+    Then,
+}
+
+impl From<ExampleLabel> for &'static str {
+    fn from(label: ExampleLabel) -> Self {
+        match label {
+            ExampleLabel::It => "It",
+            ExampleLabel::Example => "Example",
+            ExampleLabel::Then => "Then",
+        }
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct ExampleInfo {
+    pub label: ExampleLabel,
+    pub name: String,
+    pub failure: Option<String>,
+}
+
+pub struct Example<T> {
+    pub(crate) info: ExampleInfo,
```

Comment:
`pub(crate)` are used from other modules, but not exposed as public API. I'm quite certain their use is coherent, as in lower the visibility to the minimum of what's possible. I don't want the user to be able to create `Example<T>` at will, that's why `pub(crate) fn new<F>(info: ExampleInfo, f: F) -> Self` is `pub(crate)`, e.g.